### DISCUSSION TITLE
Switching from dictionary to list gives compiler error, but no detail on what file(s)

### DISCUSSION THREAD

**Discussion by Andre601** [woodpecker-ci/woodpecker#3677] [Ask a Question]
I've changed my woodpecker pipeline files from a dictionary-based setup to a list-based one, but now encounter issues.
The error I encounter in the CI is `[compiler]    yaml: line 39: did not find expected key`

I first thought it could be due to the fact that I set names containing spaces, but removing those did not change the error I'm encountering.

I've sent the files through a YAML validator and they all returned as valid YAML, so I'm at a loss on what the issue could be here exactly.
Also tried to use the cli, but it won't work, probably because the ci isn't installed on my PC (Running windows). Reports it can't find the file, no matter how I try to define it (relative using `./` or absolute with `C:...` path).

The latest iterations of all the files are found here: https://codeberg.org/Andre601/AdvancedServerList/src/branch/master/.woodpecker

It would be helpful if Woodpecker could tell you what file it found the errors with. Tho, not sure if that is possible with whatever it does to validate and compile the yaml files. If it is, then I hope it can be introduced eventually.

Finally some small info dump:

- Version used: 2.4.1
- Host: https://ci.codeberg.org
- Affected CI repository: https://ci.codeberg.org/repos/12576

**Comment by qwerty287** [comment]
Well, the `next` CLI says all workflows are valid… Not sure, but this looks like a bug, but I don't know what's the exact problem here.
Also, the `Config` tab is empty so something went wrong there too I'd say (https://ci.codeberg.org/repos/12576/pipeline/348/config)…

**Reply by Andre601** [reply]
My guess would've been that the list variant isn't supported, but the 2.4.x docs show them too, so idk.

**Comment by Andre601** [comment]
I've now completely rewritten all files (Basically deleted everything and copied everything by hand) and the workflow works again...

The only notable changes between the files are these:

- Double quotes were used instead of single quotes.
- A multi-line comment (Using `|-`) had spaces added to an empty line (missing indent)
- A multi-line comment (Using `|-`) no longer starts with `${CI_COMMIT_TAG}`

Based on this can I only assume one of the following causes to be the case:

- The CI didn't like the usage of single quotes for `name` values. Seems unlikely given YAML doesn't differenciate really between these two.
- The missing indent (Spaces) on the empty line separated the multi-line comment into a comment and separate settings(? Also seems unlikely and should've triggered on the YAML linters used).
- The CI does not like when a multi-line comment starts with a `$`. I feel like this to be the most likely cause here...

I'll leave this discussion unanswered since the cause of this problem is not yet found.

**Comment by qwerty287** [comment]
This has been solved, so I guess we can close it?
